app/src/main/database/index.test.ts

```diff
@@ -1009,4 +1009,149 @@ describe("Database Method Tests", () => {
       expect(s4.messagePreview).toBeNull();
     }
   });
+
+  it("getPendingEvents should purge stale events for nonexistent sources", () => {
```

Comment:
Ah, this test should really check that deletion has cascaded from `source.uuid` to `pending_events.source_uuid`....